### PR TITLE
Fix aten's native's folder docs.

### PR DESCRIPTION
**Pull request by khushi-411**
Fixes typo's in `aten/src/ATen/native/README.md`. The following were the fixes:
- Update string type to `c10::string_view` instead of `std::string`.
- Update the link `torch/_python_dispatcher.py`, which was broken.

**Link to docs:** https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/README.md

Thanks!

cc: @mruberry @kshitij12345

### CODE REVIEW

**Comment by khushi-411** [comment]
These failures seem non-relevant to the PR.
Link: https://github.com/pytorch/pytorch/runs/4848678112?check_suite_focus=true

File: aten/src/ATen/native/README.md [hunk 1]
```diff
@@ -61,8 +61,7 @@ signature.
 - `int`. Think about this like a Python int. This is translated into a C++ argument of type `int64_t`.
 - `float`. Think about this like a Python `float`. It is translated into a C++ argument of type `double`.
 - `bool`
-- `str`.  It is translated into a C++ argument of type `std::string`
```
Comment (kshitij12345) [inline aten/src/ATen/native/README.md:64]: I think we should also mention that `c10::string_view` is non-owning.

File: aten/src/ATen/native/README.md [hunk 2]
```diff
@@ -61,8 +61,7 @@ signature.
...
-  (but we should fix this, see https://github.com/pytorch/pytorch/issues/53546)
+- `str`.  It is translated into a C++ argument of type `c10::string_view`
```
Comment (kshitij12345) [inline aten/src/ATen/native/README.md:64]: I think we should also mention that c10::string_view is non-owning

Comment (khushi-411) [reply aten/src/ATen/native/README.md]: Ah, yes. Nice point! Thanks, Kshitij!

**Comment by samdow** [comment]
@mruberry adding from tag, feel free to send to someone else if that's not right

**Review by mruberry — APPROVED** [review]
LGTM! Thanks @khushi-411!

**Pull request closed without merging**